### PR TITLE
Remove ApiVersion parameter support in deployments cmdlets

### PR DESCRIPTION
**Pull request by Xynoclafe**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Resources/ResourceManager/Implementation/CmdletBase/ResourceManagerCmdletBase.cs [hunk 1]
```diff
@@ -66,14 +66,6 @@ public abstract class ResourceManagerCmdletBase : AzureRMCmdlet
         /// </summary>
         private SubscriptionSdkClient subscriptionSdkClient;
 
-        /// <summary>
-        /// Gets or sets the API version.
-        /// </summary>
-        [CmdletParameterBreakingChange("ApiVersion", ChangeDescription = "Parameter is being deprecated without being replaced")]
-        [Parameter(Mandatory = false, HelpMessage = "When set, indicates the version of the resource provider API to use. If not specified, the API version is automatically determined as the latest available.")]
-        [ValidateNotNullOrEmpty]
-        public virtual string ApiVersion { get; set; }
```
Comment (VeryEarly) [inline src/Resources/ResourceManager/Implementation/CmdletBase/ResourceManagerCmdletBase.cs:75]: removing parameter is breaking change and is not allowed in non-breaking change release

**Comment by VeryEarly** [comment]
removing parameter is breaking change and is not allowed in non-breaking change release, can you add it back and ignore it in the cmdlet implementation?

**Comment by Xynoclafe** [comment]
> removing parameter is breaking change and is not allowed in non-breaking change release, can you add it back and ignore it in the cmdlet implementation?

The task is specifically to remove that parameter as keeping it around while it does nothing may be confusing to users. When would the next breaking change release be?

**Comment by VeryEarly** [comment]
> > removing parameter is breaking change and is not allowed in non-breaking change release, can you add it back and ignore it in the cmdlet implementation?
> 
> The task is specifically to remove that parameter as keeping it around while it does nothing may be confusing to users. When would the next breaking change release be?

Should probably be this September.

**Comment by Xynoclafe** [comment]
Should the changes be merged into a preview branch before the breaking change release then? What is the procedure for that?

**Comment by VeryEarly** [comment]
> Should the changes be merged into a preview branch before the breaking change release then? What is the procedure for that?

It depends on whether you want to release a preview version.
For more information about breaking change please check:
https://github.com/Azure/azure-powershell/tree/master/documentation/breaking-changes

**Comment by Xynoclafe** [comment]
> > Should the changes be merged into a preview branch before the breaking change release then? What is the procedure for that?
> 
> It depends on whether you want to release a preview version.
> For more information about breaking change please check:
> https://github.com/Azure/azure-powershell/tree/master/documentation/breaking-changes

So when is a PR for a breaking change opened? And going through that repo, I assume I would need to add this to https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/upcoming-breaking-changes/Az.Resources-UpcomingBreakingChanges.md ?

**Comment by VeryEarly** [comment]
> > > Should the changes be merged into a preview branch before the breaking change release then? What is the procedure for that?
> > 
> > 
> > It depends on whether you want to release a preview version.
> > For more information about breaking change please check:
> > https://github.com/Azure/azure-powershell/tree/master/documentation/breaking-changes
> 
> So when is a PR for a breaking change opened? And going through that repo, I assume I would need to add this to https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/upcoming-breaking-changes/Az.Resources-UpcomingBreakingChanges.md ?

It's good practice to put breaking change in Az.Resources-UpcomingBreakingChanges.md .

We do not have exact date for next major release yet. Should be September or later

**Comment by VeryEarly** [comment]
close this PR now, you can reopen it if needed or you have any further questions.

**Pull request closed without merging**